docs/modules/ROOT/pages/api/upgrades.adoc

```diff
@@ -75,6 +107,25 @@ Requirements:
 
 Emits an {Upgraded} event.
 
+[.contract-item]
+[[UpgradeableComponent-upgrade_and_call]]
+==== `[.contract-item-name]#++upgrade_and_call++#++(ref self: ContractState, new_class_hash: ClassHash, selector: felt252, calldata: Span<felt252>)++` [.item-kind]#internal#
+
+Replaces the contract's class hash with `new_class_hash` and then calls `selector`
+from the upgraded context.
+
+Requirements:
+
+- `new_class_hash` must be different from zero.
+
+NOTE: The function call comes from the calling contract itself and not the account.
+
+NOTE: This behavior can also be achieved with a list of calls from an account since the {snip6} account standard supports multicall.
+An account can execute a list of calls with xref:IUpgradeable-upgrade[upgrade] being the first element in the list and the extra function call as the second.
+The extra call would be executed from the account and can't be front-ran.
```

Comment:
> This behavior can also be achieved with a list of calls

I wouldn't say that the achieved behaviour in that case will be the same. The major difference here is that with 
`upgrade_and_call` a call after upgrade will originate from the upgraded contract itself, and through the use of multicall the call will be executed from the calling contract (most likely some account)

If the upgraded account has a self-only-callable function, only `upgrade_and_call` will do the job, and the multicall will fail due to insufficient permissions

Also, I think the word **_calling_** can sound misleading in this context, because both the upgrading contract and the upgraded one make a call during the `upgrade_and_call` flow. To my mind **_upgrading_** and **_upgraded_** make more sense, what do you think?

Reply:
> I wouldn't say that the achieved behaviour in that case will be the same. The major difference here is that with
upgrade_and_call a call after upgrade will originate from the upgraded contract itself, and through the use of multicall the call will be executed from the calling contract (most likely some account)

I'll rephrase it. By behavior, I mean the upgrade + call—not the context

> Also, I think the word calling can sound misleading in this context, because both the upgrading contract and the upgraded one make a call during the upgrade_and_call flow. To my mind upgrading and upgraded make more sense, what do you think?

I don't love the premise of the suggestion. "...both the upgrading contract and the upgraded one..." sounds to me like we're talking about two different contracts. A call is coming from the contract both before and after the upgrade, but it's still coming from the same "calling contract," no?

That said, I'm okay with just "upgraded contract" :)